ingestion/src/metadata/ingestion/sink/file.py

```diff
@@ -45,25 +45,38 @@ def __init__(
         super().__init__()
         self.config = config
         self.metadata_config = metadata_config
-        fpath = pathlib.Path(self.config.filename)
-        # pylint: disable=consider-using-with
-        self.file = fpath.open("w", encoding=UTF_8)
-        self.file.write("[\n")
+        self.fpath = Path(self.config.filename)
+        # Build the path if it does not exist
+        if not self.fpath.parent.is_dir():
+            Path(self.config.filename).mkdir(parents=True, exist_ok=True)
         self.wrote_something = False
 
+        self.write_file("[\n")
+
```

Comment:
I want to make sure this makes sense:

Why is it that in `metadata.profiler.sink.file` why don't begin writing the file with `"[n"` , but we close both files with `"\n]"` ?